Typings are messed up

When importing version 2.0.3, TypeScript types are completely messed up:

1. It's impossible to use Promises as returned methods. Taking one method as example: `exists(options: any, callback?: any): void;`. The documentation says that it can be used with Promises too (and it works), but the type definition shows return type as void.
2. Types are just wrong. In the same method, the JSDoc definition says that the first argument (options) is optional, but the method signature marks it mandatory.
3. The library uses `any` as type anywhere. Options dictionaries, callbacks, etc: these are things that should be explicitly typed.
4. The library still uses JSDoc types everywhere, which should be migrated to TypeScript definitions.

tried it with the latest `typescript@3.1`, and has the problem

```
const isExist = await storage.bucket(CLOUD_STORAGE_BUCKET_NAME || '')
        .file(filePath)
        .exists(undefined);
```

and I have this problem `error TS1345: An expression of type 'void' cannot be tested for truthiness` and when I check the typings, the method returns `void`...

starting `typescript 3.1`,  they have enforce this, so the older version of `typescript` might work https://github.com/Microsoft/TypeScript/issues/26262

@welkie Your repo doesn't have enough code to repro the issue.

Try adding this:

```typescript
import { Storage } from "@google-cloud/storage";
async function listBuckets() {
    const storage = new Storage();
    const [buckets] = await storage.getBuckets();
    return buckets;
}
```

First the  `storage.getBuckets();` call will be an error because of lack of parameters:

```bash
$ tsc
src/index.ts:7:29 - error TS2554: Expected 2 arguments, but got 0.

7     const [buckets] = await storage.getBuckets();
                              ~~~~~~~~~~~~~~~~~~~~
```

So no, it doesn't work to just include the types, because even in your own example, it's finding the local `index.d.ts` first. So 1.7.2 is _entirely broken_ for TypeScript users.

--- a/package.json
+++ b/package.json
@@ -9,6 +9,7 @@
   },
   "repository": "googleapis/nodejs-storage",
   "main": "./build/src/index.js",
+  "types": "./build/src/index.d.ts",
   "files": [
     "build/src",
     "!build/src/**/*.map",

